app/lib/features/feature_base.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Features
+  class FeatureBase
+    def self.configure(config = {})
+      @configuration = new(config)
+    end
+
+    def self.configuration
+      (@configuration || configure)
+    end
+
+    def self.config
+      configuration.config
+    end
+
+    def self.enabled?
+      configuration.enabled?
+    end
+
+    def initialize(*)
+      raise NoMethodError, "#{__method__} not implemented in #{self.class}"
```

Comment:
We can use `NotImplementedError` here

Reply:
I see now that `SegmentDeletionConfig` and `AccountDeletionConfig` both have the exact same implementation. Why not just leaving everything for this class?